### ISSUE TITLE
Illegal Char in Path

### ISSUE THREAD

**Issue by dariusaurius** [maforget/Delete-RetroArch-Unneeded-Thumbnails#2]
Hi, thanks for your tool, it came at the right moment.

My setup breaks the tool, because I use a combination of local directorys and file shares (CIFS/SMB). This is the error, which is presented to me:

```
c:\RetroArch>"Delete Unneeded RetroArch Thumbnails.exe"
Using path: c:\RetroArch

Unbehandelte Ausnahme: System.ArgumentException: Illegales Zeichen im Pfad.
   bei System.IO.Path.CheckInvalidPathChars(String path, Boolean checkAdditional)
   bei System.IO.Path.Combine(String path1, String path2, String path3, String path4)
   bei Retroarch_Delete_Unneeded_Thumbnails.ThumbnailsEntry..ctor(String playlist, String romName)
   bei Retroarch_Delete_Unneeded_Thumbnails.Thumbnail.NewThumbnailEntry(String playlist, String romName)
   bei Retroarch_Delete_Unneeded_Thumbnails.PlaylistEntry..ctor(PlaylistJson PlaylistEntry)
   bei Retroarch_Delete_Unneeded_Thumbnails.Playlist..ctor(String path)
   bei Retroarch_Delete_Unneeded_Thumbnails.Program.Main(String[] args)
```

Inside my RetroArch folders are several symlinks to my NAS, this is due to the fact, that only with a workaround RetroArch is able to access shared folders, like [I have described in this post on the libretro forum](https://forums.libretro.com/t/load-content-and-add-content-from-shares-smb-etc/3237/16)

Here is a part of the DIR output of my folder:

```
09.02.2022  20:03    <SYMLINKD>     playlists [\\server\games\playlists]
21.01.2022  04:36                21 qt.conf
21.01.2022  04:36         6.005.248 Qt5Core.dll
21.01.2022  04:36         6.536.704 Qt5Gui.dll
21.01.2022  04:36         1.584.640 Qt5Network.dll
21.01.2022  04:36         6.041.600 Qt5Widgets.dll
08.02.2022  19:52    <DIR>          recordings
08.02.2022  19:51                52 RetroArch website.url
09.02.2022  19:22           106.977 retroarch.cfg
21.01.2022  04:36            33.341 retroarch.default.cfg
21.01.2022  04:36        15.153.851 retroarch.exe
08.02.2022  21:56               964 retroarch_qt.cfg
09.02.2022  01:01    <SYMLINKD>     roms [\\server\games\roms]
09.02.2022  01:01    <SYMLINKD>     saves [\\server\games\saves]
08.02.2022  19:51    <DIR>          screenshots
21.01.2022  04:36        10.511.307 SDL2.dll
08.02.2022  19:51    <DIR>          shaders
09.02.2022  01:01    <SYMLINKD>     states [\\server\games\states]
08.02.2022  19:51    <DIR>          styles
21.01.2022  04:36           122.368 swresample-3.dll
21.01.2022  04:36           538.624 swscale-5.dll
09.02.2022  01:01    <SYMLINKD>     system [\\server\games\system]
09.02.2022  01:02    <SYMLINKD>     thumbnails [\\server\games\thumbnails]
```

Playlists, roms and thumbnails are located on my NAS. 

Example of an playlist entry:

```
    {
      "path": "C:\\RetroArch\\roms\\psp\\Activision Hits Remixed.iso",
      "label": "Activision Hits Remixed",
      "core_path": "DETECT",
      "core_name": "DETECT",
      "crc32": "00000000|crc",
      "db_name": "Sony - PlayStation Portable.lpl"
    },
```

Hope you can help me out, thanks in advance

**Comment by maforget** [comment]
I've tried to recreate your setup and didn't see any problems. It doesn't seem to care that you are using a symlink. The part where you have a crash is just trying to create the path for the potential thumbnail. It takes your `C:\RetroArch\` add `thumbnails` to it. Then adds the playlist name (from the db_name entry, minus .lpl) and the romname as written under the label entry in the playlist. Some char are already replaced from the romname like &*:`<>? and replaced with _, since that is what retroarch does also for thumbnails name.

It might just be the label line in your playlist that as an illegal character that isn't changed like stated above. That error comes from the .NET Framework so all I can do is ignore this entry. But I really don't see a situation where this would happen.

**Comment by dariusaurius** [comment]
Thanks to your explanation I have narrowed it a little bit down by adding one playlist at a time and run the tool. It seems the problem is comming from the `FBNeo - Arcade Games` set. Since I'm using the whole set FBneo 1.0.0 with the according DAT files and therefore the standardized naming of the lables and filenames, everyone running this tool with the same playlist should run into the same error. Those lables are sometimes pretty weird, I can imagine that the .net methods for finding illegal chars is not as reliable as promised for the lables used by RetroArch. 

[This post at least suggest to be carefull with the method used by .net to find illegal chars](https://stackoverflow.com/a/8152503)

This is my playlist file, for reference: [FBNeo - Arcade Games.lpl](https://github.com/maforget/Delete-RetroArch-Unneeded-Thumbnails/files/8061419/FBNeo.-.Arcade.Games.lpl.txt)

I can provide you the FBneo set and DAT files, if you need them for troubleshooting

**Comment by maforget** [comment]
Your problem is with `Crouching Tiger Hidden Dragon 2003 (set 1) [Bootleg, Hack of "The King of Fighters 2001"]` because there are quotation marks inside the name. 

Note that I am not using the method you linked to, I just use Path.Combine, which in turn checks for illegal path.

I got the list of character from [this page](https://github.com/libretro/libretro-thumbnails#thumbnail-repository-layout) where some char are replaced by `_`, this is what I am doing before creating the path. Nowhere in that list is what happens if there is a `"` in the name. I tried replacing them with a `_` and removing them and the thumbnail in RetroArch doesn't appear. I tried also dragging a file on the desktop menu (F5) boxart area and that normally creates the file in the thumbnail folder with the correct name and no file is created.

So I believe that RetroArch doesn't support having quotation marks in the name. So what I did is just output a message saying that there is an illegal character in the name.

Check release v1.4.1
